src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/ThingConfigStoreAdapter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot.registry;
+
+import com.aws.greengrass.clientdevices.auth.configuration.RuntimeConfiguration;
+import com.aws.greengrass.clientdevices.auth.iot.Thing;
+import software.amazon.awssdk.utils.ImmutableMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/**
+ * Adapter from Thing domain to Thing runtime configuration.
+ * Thing configuration is stored as an opaque map to the runtime config store.
+ * <p>
+ * |---- runtime
+ * |    |---- clientDeviceThings:
+ * |          |---- thingName:
+ * |                |---- version: "..."
+ * |                |---- attachedCertificateIds: [...]
+ * </p>
+ */
+public class ThingConfigStoreAdapter {
+    private static final String ATTACHED_CERTIFICATE_IDS_KEY = "attachedCertificateIds";
+    private static final String THING_VERSION_KEY = "version";
+
+    private final RuntimeConfiguration runtimeConfig;
+
+    @Inject
+    protected ThingConfigStoreAdapter(RuntimeConfiguration runtimeConfig) {
+        this.runtimeConfig = runtimeConfig;
+    }
+
+    /**
+     * Retrieves client device thing from the runtime config store.
+     * Returns an empty optional if the thing is not available in the config store
+     *
+     * @param thingName client device Thing name
+     * @return optional of client device Thing
+     */
+    Optional<Thing> getThing(String thingName) {
+        Map<String, Object> thingDetails = runtimeConfig.getClientDeviceThing(thingName);
+        if (thingDetails.isEmpty()) {
+            return Optional.empty();
+        }
+
+        Thing thing = Thing.of((int) thingDetails.get(THING_VERSION_KEY),
+                thingName, (List<String>) thingDetails.get(ATTACHED_CERTIFICATE_IDS_KEY));
```

Comment:
don't blindly cast, check first

Reply:
+1 to what Michael said. If think the runtime config should return a shape with this already validated (validate that there is a list of string) and then `Thing.of` can have the validation for business logic and things of sorts

Reply:
I am concerned about domain logic sneaking in to the `RuntimeConfiguration`. Considering that we'll also be storing certificate metadata, and that the information we're storing may evolve over time, I hesitate to say we should perform validation there.

I'm almost inclined to say that `RuntimeConfiguration` should just get/put `Topics` and that this class should perform the mapping from `Thing` to `Topics`. Thoughts?

Reply:
I agree that domain logic should not sneak into `RuntimeConfiguration`. I refactored the adapter to deal with Topics directly and type-check (coerce) results appropriately. Also versioned the thing runtime config schema for better backward compatibility for future schema changes.